queries.sql

```diff
@@ -79,3 +79,69 @@ SELECT COUNT(*), owners.full_name
    ON animals.owner_id = owners.id
    GROUP BY owners.full_name
    ORDER BY COUNT DESC LIMIT 1;
+
+SELECT vets.name, animals.name , date_of_visit
+   FROM animals JOIN visits
+   ON animals.id = visits.animals_id
+   JOIN vets
+   ON vets.id = visits.vets_id 
+   WHERE vets.name = 'William Tatcher'
+   ORDER BY date_of_visit DESC LIMIT 1;
+
+SELECT vets.name, COUNT(visits.date_of_visit) 
+   FROM visits JOIN vets
+   ON vets.id = visits.vets_id
+   WHERE vets.name = 'Stephanie Mendez';
+   GROUP BY vets.name;
+
+SELECT vets.name, specializations.species_id, specializations.vets_id, species.name
+   FROM specializations FULL OUTER JOIN species
+   ON specializations.id = species.id
+   FULL OUTER JOIN vets
+   ON specializations.vets_id = vets.id;
+
+SELECT vets.name, animals.name , visits.date_of_visit
+   FROM animals JOIN visits
+   ON animals.id = visits.animals_id
+   JOIN vets
+   ON vets.id = visits.vets_id
+   WHERE vets.name = 'Stephanie Mendez' AND date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animals.name, COUNT(visits.date_of_visit)
+   FROM animals JOIN visits
+   ON animals.id = visits.animals_id
+   GROUP BY animals.name
+   ORDER BY COUNT DESC LIMIT 1;
+
+SELECT vets.name, animals.name , visits.date_of_visit
+   FROM vets JOIN visits
+   ON vets.id = visits.vets_id
+   JOIN animals
+   ON visits.animals_id = animals.id
+   WHERE vets.name = 'Maisy Smith'
+   ORDER BY visits.date_of_visit ASC LIMIT 1;
+
+SELECT animals.id, animals.name, animals.date_of_birth,animals.neutered,vets.name,vets.age,visits.date_of_visit
+   FROM animals JOIN visits
+   ON animals.id = visits.animals_id
+   JOIN vets
+   ON vets.id = visits.vets_id
+   ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+SELECT vets.name, COUNT(visits.date_of_visit)
+   FROM vets JOIN visits
+   ON vets.id = visits.vets_id
+   JOIN specializations
+   ON visits.vets_id = specializations.vets_id
+   WHERE specializations.species_id IS NULL
+   GROUP BY vets.name;
```

Comment:
- [ Required ] I think this query does not return a correct result. In the screenshot, the result is an empty result set which should not be the case. I would like to suggest that you refactor your code as shown below so that a correct result set is returned.

```SQL
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)
```
## Current result set
![](https://user-images.githubusercontent.com/67557999/253728552-2561b374-21a5-47f0-abe9-1874d4f5147b.png)